### PR TITLE
Enable clang implicit conversion warnings

### PR DESCRIPTION
**Pull request by QuintenJohnson**
This enables implicit conversion warnings except sign-conversion on clang builds.

### CODE REVIEW

File: minlzlib/CMakeLists.txt [hunk 1]
```diff
@@ -18,6 +18,6 @@ if(MSVC)
     set(CMAKE_C_FLAGS_RELWITHDEBINFO "/Ox /Ob2 /Oi /Ot /Oy /GF /Gy /MT /Zi /wd4214 /permissive-")
     set(CMAKE_SHARED_LINKER_FLAGS_RELWITHDEBINFO "/OPT:ICF /OPT:REF /NODEFAULTLIB /NOENTRY /DEBUG /EXPORT:XzDecode /EMITPOGOPHASEINFO /NOVCFEATURE /NOCOFFGRPINFO /FILEALIGN:512 /DRIVER /MANIFEST:NO /PDBALTPATH:minlz.pdb /MERGE:.edata=.rdata")
 else()
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror -Wno-unknown-pragmas -Wno-multichar")
-    set(CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS} -Ofast -Wall -Werror -Wno-multichar")
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror -Wconversion -Wno-sign-conversion -Wno-unknown-pragmas -Wno-multichar")
```
Comment (ionescu007) [inline minlzlib/CMakeLists.txt]: Can you fix the spacing? See the diff. Thanks!

File: minlzdec/CMakeLists.txt [hunk 2]
```diff
@@ -10,6 +10,6 @@ if(MSVC)
     set(CMAKE_C_FLAGS_RELWITHDEBINFO "/Ox /Ob2 /Oi /Ot /Oy /GF /Gy /MT /Zi /permissive-")
     set(CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO "/OPT:ICF /OPT:REF /DEBUG /EMITPOGOPHASEINFO /NOVCFEATURE /NOCOFFGRPINFO /FILEALIGN:512 /DRIVER /MANIFEST:NO /PDBALTPATH:minlzdec.pdb")
 else()
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror -Wno-multichar")
-    set(CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS} -Ofast -Wall -Werror")
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -Werror -Wconversion -Wno-sign-conversion -Wno-multichar")
```
Comment (ionescu007) [inline minlzdec/CMakeLists.txt]: Can you fix the spacing? See the diff. Thanks!

**Review by ionescu007 — CHANGES_REQUESTED** [review]
If you could please fix spacing in the CMakeLists.txt changes, I'll merge this ASAP.

File: minlzlib/CMakeLists.txt (hunk 1 above)
Comment (QuintenJohnson) [reply minlzlib/CMakeLists.txt]: Amended commit to fix.

File: minlzdec/CMakeLists.txt (hunk 2 above)
Comment (QuintenJohnson) [reply minlzdec/CMakeLists.txt]: Amended commit to fix.

**Pull request merged**